src/sql-parser/src/parser.rs

```diff
@@ -834,35 +836,33 @@ impl Parser {
                     // it takes on the precedence of those tokens. Otherwise it
                     // is not an infix operator, and therefore has zero
                     // precedence.
-                    Some(Token::Word(k)) if k.keyword == "IN" => Precedence::Cmp,
-                    Some(Token::Word(k)) if k.keyword == "BETWEEN" => Precedence::Cmp,
-                    Some(Token::Word(k)) if k.keyword == "LIKE" => Precedence::Cmp,
+                    Some(Token::Word(k)) if k.keyword == "IN" => Precedence::Like,
+                    Some(Token::Word(k)) if k.keyword == "BETWEEN" => Precedence::Like,
+                    Some(Token::Word(k)) if k.keyword == "LIKE" => Precedence::Like,
                     _ => Precedence::Zero,
                 },
                 Token::Word(k) if k.keyword == "IS" => Precedence::Is,
-                Token::Word(k) if k.keyword == "IN" => Precedence::Cmp,
-                Token::Word(k) if k.keyword == "BETWEEN" => Precedence::Cmp,
-                Token::Word(k) if k.keyword == "LIKE" => Precedence::Cmp,
+                Token::Word(k) if k.keyword == "IN" => Precedence::Like,
+                Token::Word(k) if k.keyword == "BETWEEN" => Precedence::Like,
+                Token::Word(k) if k.keyword == "LIKE" => Precedence::Like,
                 Token::Eq | Token::Lt | Token::LtEq | Token::Neq | Token::Gt | Token::GtEq => {
                     Precedence::Cmp
                 }
-                Token::Plus | Token::Minus => Precedence::Plus,
-                Token::Mult | Token::Div | Token::Mod => Precedence::Times,
-                Token::DoubleColon => Precedence::DoubleColon,
-                // TODO(jamii) it's not clear what precedence postgres gives to json operators
-                Token::JsonGet
-                | Token::JsonGetAsText
-                | Token::JsonGetPath
-                | Token::JsonGetPathAsText
-                | Token::JsonContainsJson
+                Token::JsonContainsJson
                 | Token::JsonContainedInJson
                 | Token::JsonContainsField
                 | Token::JsonContainsAnyFields
                 | Token::JsonContainsAllFields
+                | Token::JsonContainsPath => Precedence::JsonCompare,
+                Token::JsonGet
+                | Token::JsonGetAsText
+                | Token::JsonGetPath
+                | Token::JsonGetPathAsText
                 | Token::JsonConcat
-                | Token::JsonDeletePath
-                | Token::JsonContainsPath => Precedence::Json,
-
+                | Token::JsonDeletePath => Precedence::JsonOp,
```

Comment:
Unrelated, and you would know better than I, but Postgres probably has names for these ops that are more general than just JSON ops, right?